crates/aptos-rosetta/src/account.rs

```diff
@@ -137,156 +142,198 @@ async fn get_balances(
+                    version,
+                )
+                .await
+                {
+                    Ok(Some(balance_result)) => {
+                        if let Some(balance) = balance_result.balance {
+                            total_requested_balance = Some(
+                                total_requested_balance.unwrap_or_default()
+                                    + u64::from_str(&balance.value).unwrap_or_default(),
+                            );
+                        }
+                        lockup_expiration = balance_result.lockup_expiration;
+                    },
+                    result => {
+                        warn!(
+                                        "Failed to retrieve requested balance for account: {}, address: {}: {:?}",
+                                        owner_address, contract.pool_address, result
+                                    )
+                    },
+                }
             }
-
-            for currency in currencies {
+            if let Some(balance) = total_requested_balance {
                 balances.push(Amount {
-                    value: 0.to_string(),
-                    currency,
-                });
+                    value: balance.to_string(),
+                    currency: native_coin(),
+                })
             }
+
+            /* TODO: Right now operator stake is not supported
+            else if account.is_operator_stake() {
+                // For operator stake, filter on operator address
+                let operator_address = account.operator_address()?;
+                if let Some(contract) = store.staking_contracts.get(&operator_address) {
+                    balances.push(get_total_stake(
+                        rest_client,
+                        &account,
+                        contract.pool_address,
+                        version,
+                    ).await?);
+                }
+            }*/
         }
+    }
 
-        // Retrieve balances
-        Ok((
-            sequence_number,
-            maybe_operators,
-            balances,
-            lockup_expiration,
-        ))
+    // Filter currencies to lookup
+    let currencies_to_lookup = if let Some(currencies) = maybe_filter_currencies {
+        currencies.into_iter().collect()
     } else {
-        // If it fails, we return 0
-        // TODO: This should probably be fixed to check if the account exists.  Then if the account doesn't exist, return empty balance, otherwise error
-        Ok((
-            0,
-            None,
-            vec![Amount {
-                value: 0.to_string(),
-                currency: native_coin(),
-            }],
-            0,
-        ))
+        server_context.currencies.clone()
+    };
+
+    // Retrieve the fungible asset balances and the coin balances
+    for currency in currencies_to_lookup.iter() {
+        match *currency {
+            // FA only
+            Currency {
```

Comment:
Nit, extract this into its own function, or maybe a helper on the Currency type.